clang/test/CheckedC/parsing/invalid-where-clause.c

```diff
@@ -0,0 +1,31 @@
+// RUN: %clang_cc1 -fsyntax-only -verify %s
+
+// Test parsing errors for incorrect where clause.
+
+void invalid_cases(_Nt_array_ptr<char> p, int a, int b) {
+  _Where ; // expected-error {{invalid where clause}} expected-error {{expected expression}}
+  _Where ;; // expected-error {{invalid where clause}} expected-error {{expected expression}}
+  _Where _Where; // expected-error {{invalid where clause}} expected-error {{expected expression}}
+  _Where _And; // expected-error {{expected expression}} expected-error {{invalid where clause}} expected-error {{expected expression}}
+  _Where _And _And _And; // expected-error {{expected expression}} expected-error {{invalid where clause}} expected-error {{expected expression}}
+  _Where a; // expected-error {{expected comparison operator in where clause equality expression}}
```

Comment:
expected comparison operator in where clause equality expression --> expected comparison operator in equality expression